pkg/trie/triedb/db/memory_db.go

```diff
@@ -0,0 +1,82 @@
+// Copyright 2024 ChainSafe Systems (ON)
+// SPDX-License-Identifier: LGPL-3.0-only
+
+package db
+
+import (
+	"bytes"
+
+	"github.com/ChainSafe/gossamer/lib/common"
+	"github.com/ChainSafe/gossamer/pkg/trie/db"
+)
+
+type memoryDBEntry struct {
+	value []byte
+	rc    int32
```

Comment:
what is this rc?

Reply:
From what it seems it is some sort of reference counting mechanism, is this correct? I only see setting and incrementing of this, no decrementing, is this intentional? 

Aside from that, it is not clear to me why this is necessary, could you give me some context?

Reply:
Yes this is a reference counter, I based this implementation on [Parity's](https://github.com/paritytech/trie/blob/master/memory-db/src/lib.rs#L41-L92) I guess that the rc is required because we can share this DB between different tries, so its a way to prevent one trie to delete a value that another is using maybe 🤔

Reply:
I can remove it, but I think there could be some issues if we are sharing the DB between different trieDBs unless we use some type of prefix for each one 🤔

Reply:
Removed, since we are using it only for testing